### ISSUE TITLE
FP about the rule NonThreadSafeSingleton

### ISSUE THREAD

**Issue by vanguard-1024** [pmd/pmd#4073]
**Affects PMD Version:6.47.0** 

**Rule:NonThreadSafeSingleton**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.47.0/pmd_rules_java_multithreading.html#nonthreadsafesingleton>

**Description:**
Hi, we are doing research in testing static analyzer. Our approach found a false positive about the rule NonThreadSafeSingleton, PMD should focus on the line `if(buz == null)` because it can affect the multi-thread result acutally, namely it does not need to report a warning in line 4. Hence, I think this is a false positive.

**Code Sample demonstrating the issue:**
```java
public class Foo {
  private static List buz = null;
  private static List bar() {
    if (true) {  // should not report a warning in this line
      if (buz == null) {
        buz = Collections.get(Integer.MAX_SIZE);
      }
    }
    return buz;
  }
}
```

**Expected outcome:**

PMD reports a violation at line 4, but that's wrong. That's a false positive.

**Running PMD through:** *[Maven]*
